framework/set/resources/rancher2/setProvidersAndUsersTF.go

```diff
@@ -81,10 +81,10 @@ func createRequiredProviders(rootBody *hclwrite.Body, configMap []map[string]any
 		}))
 	}
 
-	if awsProviderVersion != "" && terraformConfig.Provider == defaults.Aws && customModule {
+	if cloudProviderVersion != "" && terraformConfig.Provider == defaults.Aws && customModule {
```

Comment:
This block should be fine. As mentioned in the prior comment, you need to do the exact same thing in the `linodeProviderVersion` code snippet that is below this one.